src/main/webapp/app/components/tabs/CurationHistoryTab.tsx

```diff
@@ -70,6 +70,51 @@ const CurationHistoryTab = observer(({ historyData, usersData, addUsersListener,
                 objectField: 'effect',
               });
             }
+            if (record.new['penetrance']) {
+              data.push({
+                record,
+                timeStamp: history.timeStamp,
+                admin: history.admin,
+                location: record.location,
+                objectField: 'penetrance',
+              });
+            }
+            if (record.new['inheritanceMechanism']) {
+              data.push({
+                record,
+                timeStamp: history.timeStamp,
+                admin: history.admin,
+                location: record.location,
+                objectField: 'inheritanceMechanism',
+              });
+            }
+            if (record.new['monoallelic']) {
+              data.push({
+                record,
+                timeStamp: history.timeStamp,
+                admin: history.admin,
+                location: record.location,
+                objectField: 'monoallelic',
+              });
+            }
+            if (record.new['biallelic']) {
+              data.push({
+                record,
+                timeStamp: history.timeStamp,
+                admin: history.admin,
+                location: record.location,
+                objectField: 'biallelic',
+              });
+            }
+            if (record.new['mosaic']) {
+              data.push({
+                record,
+                timeStamp: history.timeStamp,
+                admin: history.admin,
+                location: record.location,
+                objectField: 'mosaic',
+              });
```

Comment:
The content seems too similar to me. Can we just loop through an array?

Reply:
Good point. I will create a utility function that checks for one of the possible object fields, so we only need one if condition.